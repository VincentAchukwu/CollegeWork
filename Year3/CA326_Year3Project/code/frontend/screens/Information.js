import React from "react";
import { StyleSheet, Text, View, StatusBar, ScrollView, Image } from "react-native";
import Hyperlink from "react-native-hyperlink";

const Information = () => {
    return (
        <View style={styles.container}>
            <StatusBar backgroundColor="#1E90FF" barStyle="light-content"/>

            <ScrollView>
                <View style= {styles.header}>
                    <Text style= {styles.headerContent}> Introduction </Text>
                </View>
                <View style= {styles.body}>
                    <Text style= {styles.bodyContent}>
                        You might be wondering where our source of data is from and can
                        it be trusted ? We get our data from a reliable source called fixer.io
                    </Text>
                </View>

                <View style= {styles.header}>
                    <Text style= {styles.headerContent}> About Fixer </Text>
                </View>
                <View style= {styles.body}>
                    <Text style= {styles.bodyContent}>
                        Fixer is a JSON API that gives information on Foreign Exchange Rates
                        and Currency Conversion. It is updated every 10 minutes and is a trusted source
                        worldwide.
                    </Text>

                    <Image
                        style= {styles.image}
                        source= {require("../assets/images/Fixer.png")}
                    />

                    <Text style= {styles.bodyContent}>
                        To Find Out More About Fixer:
                    </Text>
                    <Hyperlink linkDefault={ true }>
                        <Text style= {{color: "blue", marginTop: "1%"}}> https://fixer.io/ </Text>
                    </Hyperlink>
                </View>
            </ScrollView>
        
        </View>
        )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: "#ffffff"
    },
    header: {
        marginTop: "6%",
        justifyContent: "center",
        alignItems: "center",
    },
    headerContent: {
        fontSize: 24,
        fontWeight: "bold",
        color: "#1E90FF",
    },
    images: {
        marginTop: "1%",
    },
    image: {
        width: 450,
        height: 350,
        resizeMode: "contain",
    },
    body: {
        justifyContent: "center",
        alignItems: "center",
        paddingLeft:"8%",
        paddingRight: "8%",
    },
    bodyContent: {
        fontSize: 16
    }
})

export default Information;