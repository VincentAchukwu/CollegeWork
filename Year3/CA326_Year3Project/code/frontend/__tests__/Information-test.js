import "react-native";
import React from "react";
import Information from "../screens/Information";
import renderer from "react-test-renderer";

test("Information Snapshot", () =>{
    const snap = renderer.create(
        <Information/>
    ).toJSON;

expect(snap).toMatchSnapshot();
})