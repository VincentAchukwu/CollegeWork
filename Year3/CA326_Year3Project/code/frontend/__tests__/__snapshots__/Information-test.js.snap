// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Information Snapshot 1`] = `[Function]`;
